Kaspersky/kaspersky_endpoint_security/ingest/parser.yml

```
@@ -0,0 +1,79 @@
+name: kaspersky-endpoint-security
+pipeline:
+  - name: parsed_event
+    external:
+      name: kv.parse-kv
+      properties:
+        value_sep: ":"
+        item_sep: \\r\\n
+
+  - name: parsed_date
+    external:
+      name: date.parse
+      properties:
+        input_field: "{{parsed_event.message.get('Release date')}}"
+        output_field: datetime
+    filter: "{{ parsed_event.message.get('Error') != None }}"
+
+  - name: set_kv_fields
+  - name: set_malware_fields
+  - name: set_not_processed_fields
+  - name: set_error_verifying_fields
+stages:
+  set_kv_fields:
+    actions:
+      - set:
+          event.kind: "event"
+          observer.vendor: "Kaspersky"
+          observer.product: "Kaspersky Endpoint Security"
+          observer.type: "edr"
+
+  set_malware_fields:
+    actions:
+      - set:
+          event.category: ["malware"]
+          event.type: ["info"]
+          event.reason: "{{ parsed_event.message.get('Reason') }}"
+          event.action: "{{ parsed_event.message.get('Result description') }}"
+          file.hash.md5: "{{ parsed_event.message.get('MD5') }}"
+          file.hash.sha256: "{{ parsed_event.message.get('SHA256') }}"
+          file.path: "{{ parsed_event.message.get('Object') }}"
+          file.name: "{{ parsed_event.message.get('Object').split(\"\\\\\")[-1] or parsed_event.message.get('Object').split(\"/\")[-1]}}"
+          user.name: "{{ parsed_event.message.get('User').split(' ')[0].split(\"\\\\\")[1] }}"
+          user.domain: "{{ parsed_event.message.get('User').split('(')[1][:-1] }}"
+          threat.software.name: "{{ parsed_event.message.get('Name') }}"
+        filter: "{{ parsed_event.message.get('Type') != None }}"
+
+      - set:
+          threat.software.type: "Malware"
+        filter: "{{ parsed_event.message.get('Type')!=None and parsed_event.message.get('Type')=='Virus' }}"
```

If you move the previous filter in the pipeline part, you can simplify this one:
```suggestion
        filter: "{{parsed_event.message.get('Type')=='Virus' }}"
```